### PR TITLE
Ensure meson's prefix is a valid absolute path

### LINKED ISSUE #17204: [bug] src\meson.build:1:0: ERROR: prefix value '/' must be an absolute path
### Describe the bug

When calling `meson.configure()` on Windows, it fails as meson considers the [`--prefix=/`](https://github.com/conan-io/conan/blob/f46c61ee40199054d1ea1b4149c58e740cccb3c0/conan/tools/meson/meson.py#L49-L50) as not being absolute on Windows. This is briefly theorized about in issue https://github.com/mesonbuild/meson/issues/12880 which is referenced in the code, but unfortunately the assumption to set / as the prefix turns out to be wrong for Windows:

```text
18:19:38  inih/58: RUN: meson setup --native-file "C:\conan_cache\b\inihdc984b83cfbdd\b\build\generators\conan_meson_native.ini" "C:\conan_cache\b\inihdc984b83cfbdd\b\build" "C:\conan_cache\inih574f2c03001fa\s\src" --prefix=/
18:19:38  conanvcvars.bat: Activating environment Visual Studio 17 - amd64 - winsdk_version=None - vcvars_ver=14.4
18:19:39  [vcvarsall.bat] Environment initialized for: 'x64'
18:19:40  [vcvarsall.bat] Environment initialized for: 'x64'
18:19:40  The Meson build system
18:19:40  Version: 1.6.0
18:19:40  Source dir: C:\conan_cache\inih574f2c03001fa\s\src
18:19:40  Build dir: C:\conan_cache\b\inihaa284fe8813d4\b\build
18:19:40  Build type: native build
18:19:40  
18:19:40  ..\..\..\..\inih574f2c03001fa\s\src\meson.build:1:0: ERROR: prefix value '/' must be an absolute path
18:19:40  
18:19:40  A full log can be found at C:\conan_cache\b\inihaa284fe8813d4\b\build\meson-logs\meson-log.txt
```

Replacing `/` with `C:\` fixes it, however! One way to achieve this is to use `f' --prefix={os.path.abspath("/")}'` instead of the plain /. Another option is [`os.path.abspath('.').split(os.path.sep)[0]+os.path.sep` from stackoverflow](https://stackoverflow.com/a/48335436), but I believe os.path.abspath should be sufficient and work on all relevant OSs.

Conan versions: Tested with 2.4.1 and 2.8.1, also checked source from develop2 branch and it's still there
OS: Windows (Container: `mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022`)
Meson: 1.6.0

### How to reproduce it

This is a (not-so-minimal-but-)working example recipe. It's a bit convoluted as I extracted the bits and pieces from our shared code base etc., but in general it's enough to run any `meson.configure()` on Windows to trigger the error.
```python
from conan import ConanFile
from conan.tools.meson import Meson, MesonToolchain
from conan.tools.cmake import cmake_layout
from conan.tools.files import collect_libs
from conan.tools.scm import Git

class Recipe(ConanFile):
    name = "inih"
    version = "58"
    url = "https://github.com/benhoyt/inih.git"
    package_type = "shared-library"
    settings = "os", "compiler", "build_type", "arch"
    options = {"fPIC": [True, False], "shared": [True, False]}

    def config_options(self):
        if self.settings.get_safe("os") == "Windows":
            self.options.rm_safe("fPIC")
        else:
            self.options.fPIC = True
        self.options.shared = True

    def generate(self):
        tc = MesonToolchain(self)
        tc.project_options["with_INIReader"] = True
        tc.project_options["distro_install"] = True
        if self.settings.os == "Windows":
            tc.project_options["default_library"] = "shared"
        tc.generate()

    def layout(self):
        cmake_layout(self, src_folder="src")

    def source(self):
        git = Git(self)
        git.clone(
            self.url,
            ".",
            ["--branch", f"r{self.version}", "--depth", "1"],
            hide_url=False,
        )

    def build(self):
        meson = Meson(self)
        meson.configure()
        meson.build()

    def package(self):
        meson = Meson(self)
        meson.install()

    def package_info(self):
        self.cpp_info.libs = collect_libs(self)
```

### PR DESCRIPTION
**Pull request by shoeffner**
The path / is not considered an absolute path on Windows, failing with:

    ERROR: prefix value '/' must be an absolute path

os.path.abspath ensures that "/" becomes a valid absolute path on all major platforms, resolving to "/" on Linux and macOS, and to "C:\\" (or whichever filesystem is currently active, could be E:\ as well) on Windows.

Closes #17204.

Changelog: Fix: Use a valid prefix path for `meson.configure()` on Windows, to avoid failures in Python 3.13.
Docs: Omit

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop2/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code. -> No, line length is longer than 80 or such
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. -> No

### CODE REVIEW

**Comment by memsharded** [comment]
Thanks for your contribution!

This PR is failing in Windows with:

```
myhello  undefined
E           
E             User defined options
E               Native files: C:\J\t_v2\9ed6\PY38\tmp_gcx6x3iconans\path with spaces\build-release\conan\conan_meson_native.ini
E               prefix      : C:\
E           
E           Found ninja-1.8.2 at "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja\ninja.EXE"
E           
E           conanfile.py (demo/0.1): Meson build cmd: meson compile -C "C:\J\t_v2\9ed6\PY38\tmp_gcx6x3iconans\path with spaces\build-release" -j16
E           conanfile.py (demo/0.1): RUN: meson compile -C "C:\J\t_v2\9ed6\PY38\tmp_gcx6x3iconans\path with spaces\build-release" -j16
E           conanvcvars.bat: Activating environment Visual Studio 15 - amd64 - winsdk_version=None - vcvars_ver=14.1
E           [vcvarsall.bat] Environment initialized for: 'x64'
E           [1/2] Compiling C object myhello.exe.p/src_main.c.obj
E           FAILED: myhello.exe.p/src_main.c.obj 
E           "cl" "-Imyhello.exe.p" "-I." "-I.." "-DNDEBUG" "/MD" "/nologo" "/showIncludes" "/utf-8" "/W2" "/O2" "/Gw" "/Fdmyhello.exe.p\src_main.c.pdb" /Fomyhello.exe.p/src_main.c.obj "/c" ../src/main.c
E           .\config.h(8): error C2001: newline in constant
E           ninja: build stopped: subcommand failed.
E           INFO: autodetecting backend as ninja
```

It is possible that the ``os.path.abspath`` result needs to be back to forward slash for Meson to process it?

**Comment by shoeffner** [comment]
Thanks for the logs! I don't think it's the \ vs / since the native files also use backslashes – and I was able to build inih (see our discussion in #17204) with this change.

I will try to get the full test suite up and running on our build machine and see if I can reproduce and fix the issue.

**Comment by memsharded** [comment]
> I will try to get the full test suite up and running on our build machine and see if I can reproduce and fix the issue.

Probably not worth the effort. The full suite, specially the ``functional`` tests, have tons of tools to configure in ``conftest`` and ``conftest_user.py``. You are probably good with running just a small subset. The test failing is ``test_meson_using_prefix_path_in_application``, probably running tests in ``functional/toolchain/meson`` is more than enough.

**Comment by shoeffner** [comment]
**tl;dr:** `C:\` is interpreted by the failing test as a `newline in constant`:
```
E           .\config.h(8): error C2001: newline in constant
```
Switching it to `C:/` works, and all other test results remain unchanged.

**long read:**

Okay, that took longer than I expected. I had to change test\utils\tools.py as our mvsc has version 194 (in contrast to the default test version 191) and it took me a while to figure that out.

Here's the result of the unmodified (except for the version as described above) develop2 branch with pip-installed meson:
```
C:\src\conan>pytest -n 16 test\functional\toolchains\meson
===================================== test session starts ======================================
platform win32 -- Python 3.10.11, pytest-7.4.4, pluggy-1.5.0
rootdir: C:\src\conan
configfile: pytest.ini
plugins: xdist-3.6.1
16 workers [32 items]
ssssssssssssssss..s..s.....F....                                                          [100%]
=========================================== FAILURES ===========================================
___________________________________ test_meson_lib_template ____________________________________
===================== 1 failed, 13 passed, 18 skipped in 74.57s (0:01:14) ======================
```
I haven't checked yet why this particular test fails, maybe it's because I changed the compiler version locally and this assumes a specific hash?

This is the (shortened) output with the choco-installed meson:
```
=================================== short test summary info ====================================
FAILED test/functional/toolchains/meson/test_install.py::MesonInstall::test_install - Exception:
FAILED test/functional/toolchains/meson/test_subproject.py::test_subproject - Exception:
FAILED test/functional/toolchains/meson/test_backend.py::test_cross_x86 - Exception:
FAILED test/functional/toolchains/meson/test_meson_and_gnu_deps_flags.py::TestMesonToolchainAndGnuFlags::test_mesondeps_flags_are_being_appended_and_not_replacing_toolchain_ones - Exception:
FAILED test/functional/toolchains/meson/test_v2_meson_template.py::test_meson_lib_template - Exception:
FAILED test/functional/toolchains/meson/test_cross_compilation.py::test_windows_cross_compiling_x86 - Exception:
FAILED test/functional/toolchains/meson/test_meson.py::test_meson_using_prefix_path_in_application - Exception:
FAILED test/functional/toolchains/meson/test_meson.py::MesonToolchainTest::test_definition_of_global_options - Exception:
FAILED test/functional/toolchains/meson/test_v2_meson_template.py::test_meson_exe_template - Exception:
FAILED test/functional/toolchains/meson/test_meson.py::MesonToolchainTest::test_meson_default_dirs - Exception:
FAILED test/functional/toolchains/meson/test_pkg_config_reuse.py::MesonPkgConfigTest::test_reuse - Exception:
FAILED test/functional/toolchains/meson/test_test.py::MesonTest::test_reuse - Exception:
FAILED test/functional/toolchains/meson/test_preprocessor_definitions.py::MesonPreprocessorDefinitionsTest::test_build - Exception:
========================== 13 failed, 1 passed, 18 skipped in 20.62s ===========================
```
And the error is exactly what I observed initially:
```
E           ..\meson.build:2:0: ERROR: prefix value '/' must be an absolute path
```

With my fix, I can reproduce the error observed in the CI pipeline:
```
FAILED test/functional/toolchains/meson/test_meson.py::test_meson_using_prefix_path_in_application - Exception:
```
A related issue seems to be https://github.com/conan-io/conan/issues/14213 and the test there checks if the prefix is set correctly. It appears that particular test catches problematic backslashes in the prefix path, so `C:\`, while valid for meson, might be a broken value for some recipes. Using a unix filesep as described in the issue works, i.e., passing `C:/` instead.

I updated the code and added a `_prefix` property for which I documented the constraints and reasoning why the os.path.abspath and replace are required.

**Review by memsharded — APPROVED** [review]
Many thanks for your contribution @shoeffner 

Checked that without this the test suite fails in Python 3.13.
We might need to add it to our CI, it seems that 3.13 is changing a few things that are breaking.

Lets merge this to include it in next Conan 2.9 asap
